Makefile

```diff
@@ -0,0 +1,484 @@
+
+
+android-armv7.stamp: android-env
+	$(CARGO) rustc --locked \
+	  --target armv7-linux-androideabi \
+	  --package core-crypto-ffi \
+	  --crate-type=cdylib --crate-type=staticlib \
+	  --release -- -C strip=symbols
+	touch $@
+
+.PHONY: android-armv7
+android-armv7: android-armv7.stamp
+
+android-armv8.stamp: android-env
+	$(CARGO) rustc --locked \
+	  --target aarch64-linux-android \
+	  --package core-crypto-ffi \
+	  --crate-type=cdylib --crate-type=staticlib \
+	  --release -- -C strip=symbols
+	touch $@
+
+.PHONY: android-armv8
+android-armv8: android-armv8.stamp
+
+android-x86.stamp: android-env
+	# Link clang_rt.builtins statically for x86_64 Android
+	$(CARGO) rustc --locked \
+	  --target x86_64-linux-android \
+	  --package core-crypto-ffi \
+	  --crate-type=cdylib --crate-type=staticlib \
+	  --release -- \
+	  -C strip=symbols \
+	  -l static=clang_rt.builtins-x86_64-android \
+	  -L $$(dirname $$($(ANDROID_NDK_HOME)/toolchains/llvm/prebuilt/$(PLATFORM_DIR)/bin/clang --print-runtime-dir))/linux
+	touch $@
+
+.PHONY: android-x86
+android-x86: android-x86.stamp
+
+android.stamp: android-armv7.stamp android-armv8.stamp android-x86.stamp
+	touch $@
+
+.PHONY: android
+android: android.stamp
+
+.PHONY: android-test
+android-test: android
+	cd crypto-ffi/bindings && \
+	./gradlew android:build -x lint -x lintRelease
+
+#-------------------------------------------------------------------------------
+# 8) JVM native builds (Darwin + Linux)
+#-------------------------------------------------------------------------------
+
+# darwin build
+jvm-darwin.stamp: bindings-kotlin-jvm.stamp
+	cd crypto-ffi && \
+	$(CARGO) rustc --locked \
+	  --target aarch64-apple-darwin \
+	  --package core-crypto-ffi \
+	  --crate-type=cdylib --crate-type=staticlib \
+	  --release -- -C strip=symbols
+	touch $@
+
+.PHONY: jvm-darwin
+jvm-darwin: jvm-darwin.stamp
+
+# linux build
+jvm-linux.stamp: bindings-kotlin-jvm.stamp
+	cd crypto-ffi && \
+	$(CARGO) rustc --locked \
+	  --target x86_64-unknown-linux-gnu \
+	  --package core-crypto-ffi \
+	  --crate-type=cdylib --crate-type=staticlib \
+	  --release -- -C strip=symbols
+	touch $@
+
+.PHONY: jvm-linux
+jvm-linux: jvm-linux.stamp
+
+.PHONY: jvm
+ifeq ($(UNAME_S),Linux)
+jvm: jvm-linux
+else ifeq ($(UNAME_S),Darwin)
+jvm: jvm-darwin
+else
+$(error Unsupported host platform for jvm: $(UNAME_S))
+endif
+
+.PHONY: jvm-test
+jvm-test: jvm
+	cd crypto-ffi/bindings && \
+	./gradlew jvm:build -x lint -x lintRelease
+
+
+#-------------------------------------------------------------------------------
+# 9) Aggregate targets
+#-------------------------------------------------------------------------------
+
+.PHONY: mobile wasm all
+mobile: android ios
+wasm: bindings-js
+all: mobile wasm docs
+
+# ────────────────────────────────────────────────────────────────────────
+# TypeScript / JS tasks (from crypto-ffi/bindings/js/package.json)
+# ────────────────────────────────────────────────────────────────────────
```

Comment:
We're not using the same character for making horizontal separators, please compare this to e.g. `9) Aggregate targets` above. I don't have a preference either way, but we should be consistent.